### DISCUSSION TITLE
64-bit time function takes a lot of code space

### DISCUSSION THREAD

**Discussion by abarnas-airly** [zephyrproject-rtos/zephyr#59271] [Q&A]
Hello 
I am trying to optimize code size (without compiler optimization) and I noticed that 11% of my 128KB code space is taken by 64bit zephyr function converting time to ticks:

address | size (HEX) | name | size(DEC)
-- | -- | -- | --
08006d95 l     F text | 000004bc | k_us_to_ticks_ceil64 | 1212
0800a251 l     F text | 000004bc | k_us_to_ticks_ceil64 | 1212
0800eadd l     F text | 000004bc | k_us_to_ticks_ceil64 | 1212
0800f451 l     F text | 000004bc | k_ticks_to_us_floor64 | 1212
08001405 l     F text | 000004b8 | k_ms_to_ticks_ceil64 | 1208
080018bd l     F text | 000004b8 | k_ticks_to_ms_floor64 | 1208
080037c1 l     F text | 000004b8 | k_ms_to_ticks_ceil64 | 1208
08003c79 l     F text | 000004b8 | k_ticks_to_ms_floor64 | 1208
08008755 l     F text | 000004b8 | k_ms_to_ticks_ceil64 | 1208
0800c035 l     F text | 000004b8 | k_ms_to_ticks_ceil64 | 1208
0800d5a5 l     F text | 000004b8 | k_ms_to_ticks_ceil64 | 1208
0800ef99 l     F text | 000004b8 | k_ticks_to_ms_floor64 | 1208

Is it possible to fix it using some config?

My prj.conf:
```
CONFIG_GPIO=y
CONFIG_SENSOR=y
CONFIG_SERIAL=y
CONFIG_LOG=y
CONFIG_NO_OPTIMIZATIONS=y
CONFIG_MAIN_STACK_SIZE=2048
CONFIG_UART_ASYNC_API=y
CONFIG_SENSOR_LOG_LEVEL_ERR=y
CONFIG_THREAD_NAME=y
```

**Comment by ndrs-pst** [comment]
@abarnas-airly, What about the code size of 64-bit time function when `CONFIG_SIZE_OPTIMIZATIONS=y` ?.
My understanding is that Zephyr doesn't currently offer a way to finely adjust optimization settings for individual code segments. 

As a workaround, I've tweaked the CMakeLists.txt in my application project. This allows me to specifically override the COMPILE_FLAGS for the application source files, effectively setting them to `-O0`. Here’s the modification I made:
```
set_source_files_properties(
    ${APP_SRC_FILES}
    PROPERTIES
    COMPILE_FLAGS "-O0"
)
```
